homestar-runtime/tests/network/dht.rs

```diff
@@ -0,0 +1,983 @@
+        .arg("tests/fixtures/test_dht5.toml")
+        .arg("--db")
+        .arg(DB1)
+        .stdout(Stdio::piped())
+        .spawn()
+        .unwrap();
+
+    let ws_port1 = 7984;
+    if wait_for_socket_connection(ws_port1, 1000).is_err() {
+        let _ = kill_homestar(homestar_proc1, None);
+        panic!("Homestar server/runtime failed to start in time");
+    }
+
+    tokio_test::block_on(async {
+        let ws_url = format!("ws://{}:{}", Ipv4Addr::LOCALHOST, ws_port1);
+        let client = WsClientBuilder::default()
+            .build(ws_url.clone())
+            .await
+            .unwrap();
+
+        let mut sub1: Subscription<Vec<u8>> = client
+            .subscribe(
+                SUBSCRIBE_NETWORK_EVENTS_ENDPOINT,
+                rpc_params![],
+                UNSUBSCRIBE_NETWORK_EVENTS_ENDPOINT,
+            )
+            .await
+            .unwrap();
+
+        let homestar_proc2 = Command::new(BIN.as_os_str())
+            .env(
+                "RUST_LOG",
+                "homestar=debug,homestar_runtime=debug,libp2p=debug,libp2p_gossipsub::behaviour=debug",
+            )
+            .arg("start")
+            .arg("-c")
+            .arg("tests/fixtures/test_dht6.toml")
+            .arg("--db")
+            .arg(DB2)
+            .stdout(Stdio::piped())
+            .spawn()
+            .unwrap();
+
+        let ws_port2 = 7985;
+        if wait_for_socket_connection(ws_port2, 1000).is_err() {
+            let _ = kill_homestar(homestar_proc2, None);
+            panic!("Homestar server/runtime failed to start in time");
+        }
+
+        let ws_url2 = format!("ws://{}:{}", Ipv4Addr::LOCALHOST, ws_port2);
+        let client2 = WsClientBuilder::default()
+            .build(ws_url2.clone())
+            .await
+            .unwrap();
+
+        let mut sub2: Subscription<Vec<u8>> = client2
+            .subscribe(
+                SUBSCRIBE_NETWORK_EVENTS_ENDPOINT,
+                rpc_params![],
+                UNSUBSCRIBE_NETWORK_EVENTS_ENDPOINT,
+            )
+            .await
+            .unwrap();
+
+        // Poll for connection established message
+        loop {
+            if let Ok(msg) = sub1.next().with_timeout(Duration::from_secs(30)).await {
+                let json: serde_json::Value =
+                    serde_json::from_slice(&msg.unwrap().unwrap()).unwrap();
+
+                if json["type"].as_str().unwrap() == "network:connectionEstablished" {
+                    break;
+                }
+            } else {
+                panic!("Node one did not establish a connection with node two in time.")
+            }
+        }
+
+        // Run test workflow on node one
+        let _ = Command::new(BIN.as_os_str())
+            .arg("run")
+            .arg("-p")
+            .arg("9784")
+            .arg("-w")
+            .arg("tests/fixtures/test-workflow-add-one.json")
+            .output();
+
+        // TODO Should not fail with this timeout present
```

Comment:
Updating this to bring the timeout back, with an explanation comment. The idea is that we want node two to use the provider mechanism because of a timeout, not because workflow info was missing from the DHT. So we add a brief timeout to let this happen.